lib/redis/lock.rb

```diff
@@ -38,18 +38,19 @@ def lock(&block)
       try_until_timeout do
         expiration = generate_expiration
         # Use the expiration as the value of the lock.
-        break if redis.setnx(key, expiration)
+        break if redis.setnx(key, expiration) && Redis.next.setnx(key, expiration)
```

Comment:
Hmm, this becomes problematic at the 2nd deploy when you're rolling off OLD redis. Some unicorns and sidekiqs will be slower to restart than others. They'll be setting locks in NEW redis regardless of their expiration status.